### ISSUE TITLE
CLI 'create-user' option '--noemail' does not work

### ISSUE THREAD

**Issue by PitWenkin** [virtualmin/virtualmin-gpl#372]
When using the [CLI to add users](https://www.virtualmin.com/documentation/developer/cli/create_user/) I expect the followiing command to **only** create a mail user without creating an email alias:

`virtualmin create-user --domain example.com --user test --pass 123 --noemail`

However it creates a user `test` in the domain`example.com` **and** a mail alias `newuser` with directs to `test.example` (missing `.com`)

**Comment by jcameron** [comment]
What do you mean by a mail alias in this context? Normally one won't get created for a user without email enabled ..

**Comment by PitWenkin** [comment]
Mail alias as in the screenshot
![Screenshot 2022-02-15 at 07 27 38](https://user-images.githubusercontent.com/45786226/154005547-b2e8f8cb-784a-4b8b-ac99-f5ef53004a65.png)


I use the command above to to create a user (on server which have email enabled), which functions as a email account/mailbox (accessible via POP3/IMAP).

`virtualmin create-user --domain example.com --user test --pass 123` should create an account with an email adress `text@example.com`, where the emails are store in an account `test.example` which can be fetched via  POP3/IMAP.

Using `--noemail` prevented the creation of the actual adress and only created the account. I then could create any email alias/adress which pointed to the account. Alias `testing@example.com` directing to account/mailbox `test.example`

**Comment by jcameron** [comment]
That's expected actually .. the `--noemail` flag only prevents creation of the default email address, but one can be added later.

**Comment by PitWenkin** [comment]
So the fact that it does create an alias `newuser` directing to a non existing mailbox to a is a bug?

**Comment by jcameron** [comment]
That seems odd ... is the name of the alias actually "newuser" ?

**Comment by PitWenkin** [comment]
Exactly. I had this bug on serveral installations (over the last 2 years) and never found the source/ believed it was the way I used the CLI. 

`virtualmin create-user --domain example.com --user test --pass 123 --noemail` is the minimal command with which I can recreate it consistently.

Test Server is configured as follows:

- Operating system Ubuntu Linux 20.04.3
- Webmin version 1.984 
- Usermin version 1.834
- Virtualmin version 6.17-3
- Authentic theme version 19.84.7 

Global enables Features and Plugins

- Administration user	Core	6.17
- Home directory	Core	6.17
- Apache website	Core	6.17
- Apache SSL website	Core	6.17
- MySQL database	Core	6.17
- Mail for domain	Core	6.17
- Spam filtering	Core	6.17
- Log file rotation	Core	6.17	
- Webalizer reporting	Core	6.17	
- Webmin login	Core	6.17
- AWstats reporting	Plugin	5.11	
- DAV Login	Plugin	3.13	
- Protected web directories	Plugin	3.0	
- ProFTPD virtual FTP	Core	6.17	

Virtual Serverv (one and only) has only the followling:

- Quota unlimited

Enabled features:

- Apache website
- Apache SSL website
- MySQL database
- Mail for domain


I had the bug un Ubuntu 16 an 18. On virtual server with spam and virus filtering enabled.

**Comment by PitWenkin** [comment]
I just retested the whole procedure and noticed that the email address/alias `newuser@example.com` has been added to the mailbox as _additional email address_, which should only happen if I use `--extra newuser@example.com`?

Screenshot with faked domainnames:
![Screen Shot 2022-02-16 at 20 53 38](https://user-images.githubusercontent.com/45786226/154345783-dbbb8c88-6890-4447-b69c-8eb8c2833c8d.png)

**Comment by jcameron** [comment]
Yes, that should only happen if the `--extra` flag is given. 

If you create a second user, is his additional email address also the same?

**Comment by PitWenkin** [comment]
I created a second user `test2` via the exact same command:

- The _additional email address_ has been removed from the `test` user
- The _additional email address_ `newuser@example.com` has been added to the `test2` user
- The _alias_ `newuser` now direct to `test2`

**Comment by PitWenkin** [comment]
I have not configured anything I could remember. I compared `/etc/webmin/useradmin/config`on two servers and both have the same content (alphabetical sorting due comparison):

```
alias_check=0
base_gid=1000
base_uid=1000
default_group=users
default_other=1
delete_only=0
delete_root=0
display_max=400
display_mode=1
email_type=0
empty_mode=0
extra_real=0
from_files=0
gid_calc=0
gid_mode=1
group_file=/etc/group
gshadow_file=/etc/gshadow
home_base=/home
home_style=0
homedir_perms=0750
last_show=0
lock_string=*
md5=1
membox=0
new_user_gid=0
new_user_group=0
passwd_dict=0
passwd_file=/etc/passwd
passwd_progmode=0
passwd_same=0
passwd_stars=0
random_password=0
secmode=0
selinux_con=user_u:object_r:user_home_dir_t
shadow_file=/etc/shadow
shells=fixed,passwd,shells
skip_md5=0
sort_mode=0
uid_calc=0
uid_mode=1
user_files=/etc/skel
```

**Comment by jcameron** [comment]
I just remembered another place in Virtualmin where aliases for new users might be set. Check at System Settings -> Server Templates -> Default Settings -> Mail for domain -> Mail forwarders for new users.

**Comment by PitWenkin** [comment]
Nope. Already checked that

It is set to `none` on every server. If a server has a second template defined (which is never used) it is either set to `none` too, or to `From default settings`.

The only settings on that screen that are not out of the box (on one server!) are:

- `Default quota for mail users` which is set to `50MB`
- `Format for username that include domain`which is set to `username@domain`


I also checked Server Configuration -> Email Settings, but there everything is blank too.
Under Edit Users there is a button `Edit New User Default`which seems to show the same quota and forward settings as the server template. But nothing there as well.
![Screenshot 2022-02-18 at 09 17 51](https://user-images.githubusercontent.com/45786226/154644595-c8b030fb-b624-44e1-8190-d4418a250907.png)
![Screenshot 2022-02-18 at 09 20 33](https://user-images.githubusercontent.com/45786226/154644992-9e04abda-8659-4914-9b5f-312c217da9c7.png)

No server has the option enabled to sent a welcome (or update email) to new accounts

Spam filtering is set to autodelete spam older than 30 days. And I created a job on one(!) server to do the same for trash.
